test/integration/test/webpush_test.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+
+const chakram = require('chakram'), expect = chakram.expect;
+const Config = require('config-js');
+
+var config = new Config('./test/integration/lib/config/foxbox.js');
+var Prepper = require('../lib/testPrepper.js');
+
+Prepper.makeSuite('Test Push Service locally',function(){
+  var header = new Config('./test/integration/lib/config/header.js');
+  var credential = config.get('credential'); 
+
+  var setupURL = config.get('foxbox.url') + '/users/setup';
+  var loginURL = config.get('foxbox.url') + '/users/login';
+  var serviceURL = config.get('foxbox.url') + '/api/v1';
+  var serviceListURL = serviceURL + '/services'; 
+  var getterURL = serviceURL + '/channels/get';
+  var setterURL = serviceURL + '/channels/set';
+    
+  it('Create a new credential',function(){
+
+    return chakram.post(setupURL,credential)
+    .then(function(setupResp){
+      expect(setupResp).to.have.status(201);
+    });
+  });
+
+  describe ('Once logged in', function () {
+    
+    before(function() {
+      var encoded_cred = new Buffer(credential.username+
```

Comment:
I believe we have to put the credential logic in one single place. If the logic changes, we'll end up with 4 different places to update.